src/core/client.ts

```diff
@@ -417,3 +418,34 @@ function createAbortControllerFromSignal(signal?: AbortSignal) {
     else sig.addEventListener("abort", abort);
     return { abort, signal: abortController.signal };
 }
+
+function validateSignal(
+    method: string,
+    payload: Record<string, unknown>,
+    signal: AbortSignal,
+) {
+    // We use a very simple heuristic to check for AbortSignal instances
+    // in order to avoid doing a runtime-specific version of `instanceof`.
+    if (typeof signal.addEventListener === "function") {
```

Comment:
Here, if null is given as signal, it would throw a runtime error when we try to access the property.

Do we want to throw the default error message, or would we prefer to also provide the custom error message below for null signals?

Reply:
Oh I see. I will adjust this to handle `null`. Good idea, thanks!